### ISSUE TITLE
📚[DOCS]: Order of @Selector() functions matters with joined selectors and setting injectContainerState: false

### ISSUE THREAD

**Issue by rwlnd** [ngxs/store#1340]
### Description

Hi, I'm not really sure if this is `injectContainerState` related or maybe I'm just pointing out the obvious here but I just spent quiiite some time figuring out why my **joined selectors** did not resolve the correct data.

I wanted to optimize a joined `@Selector()` because it was being recalculated on EVERY state change which was completely unnecessary. So, I came across the `injectContainerState : false` section in the documentation ( https://www.ngxs.io/concepts/select#joining-selectors ) and decided to give it a go. Heck, _like that I'll even be ngxs-4 compliant already_ I was thinking. So I put `injectContainerState : false` on my configuration and refactored my selectors to only inject the state if necessary and inject other selectors.

I noticed that in the joined selector I always got the FULL state injected and not a specific slice. Which caused some unwanted errors of course.

Fast forward after quite some investigation, and trying to recreate the error in a stackblitz for the github issues section I found out that the weird behavior was because the selectors that I was injecting were declared AFTER the joined selector.

Moving the injected selector above the joined selector made the selectors act as expected.

So, as already said, maybe I'm stating the obvious here but, I think it would be helpful to have a **warning about this behavior in the documentation**. Theoretically, ordering your selectors based on dependency makes sense but I think that this will be an issue a lot of current ngxs using people will run into when they refactor towards the new default behavior of not injecting the state by default as the first parameter. While refactoring selector functions that relied on the state being injected you will not immediately think about re-arranging your selector functions.

tldr; It would be nice to add awareness of the importance of the the order of @Selector functions when using joined selectors.

## 🔬 Minimal Reproduction

I recreated the behavior in stackblitz: https://stackblitz.com/edit/angular-pax1fo
In the solution, moving the joined selector in `article.state.ts` below the 2 other selector will make this solution work as expected.

## 🌍  Your Environment
"@ngxs/store": "3.5.0"

I really hope this will at least help some people who will run into this behavior in the meanwhile.

**Comment by arturovt** [comment]
Would you like to contribute and make a PR?

**Comment by arturovt** [comment]
I have read your issue more thoughtfully and I have to say that the order of selectors does matter and it’s already mentioned in our docs. See https://www.ngxs.io/concepts/select#the-order-of-interacting-selectors

**Comment by rwlnd** [comment]
Hi arturovt, sorry for the late feedback. At the time I posted this, the docs section about the order of interacting selectors was not posted on the main docs on ngxs.io yet. Good to see it's there now, I also stumbled on it back then after doing a more extensive search in the already closed issues but forgot to update my issue. Thank you.

**Comment by markwhitfeld** [comment]
@rwlnd I have merged a PR that should fix this issue. Please could you test with the `@ngxs/store@dev` package and confirm if it is fixed for you or not.

**Comment by rwlnd** [comment]
Hi @markwhitfeld - sorry for the late feedback but I just tested this in a fork of my original stackblitz (@ngxs/stor v3.6.1) and it worked. I really think this will eliminate a lot of confusion, thanks for the effort!
